### DISCUSSION TITLE
Forceful Name Change

### DISCUSSION THREAD

**Discussion by Technetium1** [noisetorch/NoiseTorch#332] [General]
Unfortunately, lawl has demanded we stop using the name NoiseTorch ( https://github.com/noisetorch/NoiseTorch/issues/328 ). Rather than poison-pill ourselves after spending so much time auditing code, let's devise another name that can stick to the project. 

Almost anything goes, but be aware that it should be SEO-friendly, unique (within reason), and non-offensive. Remember the name will show in package managers -- so let's do better than `noise-suppression-for-voice`.
 - You can search for existing name collisions on [Repology](https://repology.org/projects/) and [pkgs.org](https://pkgs.org/)
 - You can search for trademark overlaps on [Trademarkia](https://www.trademarkia.com/) and [U.S. Patent and Trademark Office's lookup tool](https://www.uspto.gov/trademarks/search)

The sooner this happens the better, so we can redirect CI-built updates to the new URL and make the transition easier for current users.

---

**AUR MAINTAINER NOTE:**
Please resubmit your package and make a request to the mailing list to have packages merged so that your votes/comments remain intact - see: [1](https://bbs.archlinux.org/viewtopic.php?id=199094) & [2](https://bbs.archlinux.org/viewtopic.php?id=161679)

**Comment by TheDukeofErl** [comment]
> The sooner this happens the better, so we can redirect CI-built updates to the new URL and make the transition easier for current users.

In regards to this, I believe that as long as the repository has it's name changed or is just transferred things may be a lot smoother, as Github should automatically link to the new page, including the releases page/latest (e.g. https://github.com/wwmm/pulseeffects/releases going to easyeffects). This may mean that users on a version that points at a different name for the repo will still just get an update as expected, so long as the private and public keys do not require a change when this is done.

**Comment by Technetium1** [comment]
I left comments about this on [the AUR packages](https://aur.archlinux.org/packages?K=noisetorch).

**Reply by Scrumplex** [reply]
Thanks for the heads-up. I don't follow this repo much. Once there is a new name I will just push the AUR package there and then merge the current one into it

**Comment by AXDOOMER** [comment]
Please note that changing the project's name (which we have all the rights to keep) will not solve the license issues as the additional terms to Section 7, subsection `c` will stay in the license, meaning that you'll still have to abide by the additional terms when packaging for e.g. AUR.

**Comment by dahsameer** [comment]
NoiseShade

**Reply by Scrumplex** [reply]
Maybe NoiseFlare, as a synonym to NoiseTorch

**Reply by ZyanKLee** [reply]
DinFlare ?

**Reply by LEONIDAS-0117** [reply]
> Maybe NoiseFlare, as a synonym to NoiseTorch

As a current user of this project, this name is the best one.
Not only requires minimal changes to the current one (only one word needs to change), but it makes it easier to search for someone who is not aware of the name change.
Also it makes it easier to find in any search engine or package manager (noise being the main word to search for).
And for the Flare part, i think is a good way to keep the legacy of the previous name.
[The same but different.](https://www.youtube.com/watch?v=7tTfL-DtpXk)

**Reply by Technetium1** [reply]
I'd like to add that off-GitHub recommendations have also mentioned `NoiseFlare`.

**Reply by luni3359** [reply]
I got to agree it's pretty nice

**Comment by TheCapsLock** [comment]
noisezap ; noisezapper

**Comment by Technetium1** [comment]
Some root word suggestions from different platforms so far:

- Hush
- Shush
- Tone
- Silence
- Suppress
- Flare

**Comment by RiemaruKarurosu** [comment]
Do we need a new logo?

**Reply by TheDukeofErl** [reply]
I don't think that'll be an issue. I highly doubt that lawl, questionably enacting a section in the GPL, would be able to lay a copyright claim on something that's CC-BY 4.0.

**Reply by Technetium1** [reply]
If the image is [CC-BY 4.0](https://creativecommons.org/licenses/by/4.0/) then we don't need to change it.

**Comment by bursterburster** [comment]
VoiceTorch?

**Comment by RiemaruKarurosu** [comment]
NovaTorch

Notron (Noise To ReductiON)

**Comment by The-Munk** [comment]
`Miq`, like "mic", but with a Q (for quiet)
Not wildly clever, but simple and easy to remember.

**Reply by Technetium1** [reply]
Would probably have to be modified as that already exists many places https://www.trademarkia.com/trademarks-search.aspx?tn=miq

**Comment by GANOOfield** [comment]
'Noise-Outer', if anyone get the Harry Potter reference, 'Nouter' for short

**Comment by jmanecke** [comment]
NoiseFloor
NoiseClean

Trying to think of a title that highlights how well it removes noise like typing but keeps voices. This is different than just putting  a floor on background noise.
SmartNoise
SmartGate
TypeTorch
CleanVoice

**Reply by georgetasker** [reply]
CleanVoice has a nice ring to it.

**Comment by georgetasker** [comment]
How about **NoiseCrunch**.
Or maybe something more aggressive like **NoiseSmash** or **NoiseKill**.

Also for a logo how about something phoenix like which is rising out of the ashes where the ashes clearly show the word "NoiseTorch" in them as a nod to the origins of this app.

**Reply by georgetasker** [reply]
After some more thinking I've come up with some more.

**NoiseMuffler
NoiseStrangler
NoiseChoker
NoiseWhisperer**

Alternatively we could move away from the word "noise" completely as a prefix to any name.

**Comment by mbenlioglu** [comment]
Additional terms introduced by lawl are not valid and can be removed as it appears it's done so without the permission of other copyright holders i.e. contributors. This requirement is clearly stated in the license. Unless lawl:
   1. Has written permission from the contributors that made their contributions before the commit 8b054c20a909c9cc459e87ee8b56226c0428201b, or 
   2. Can provide a CLA signed by the said contributors that made them waive their copyrights
 
these terms are void and even unlawful to introduce in the first place.

**Reply by AXDOOMER** [reply]
Are you a lawyer? The question that needs to be answered is if it still applies to code Lawl contributed after changing the terms, because Lawl agreed with the terms. It's only void for previous contributors that didn't agree and contributed before the license change, you can still use their code under the old license.

**Reply by mbenlioglu** [reply]
No I'm not, but this is a very fundamental topic, yet it comes up frequent enough to know how things work. 

Let's cover the basics. Since there are no individual notices on each file, we can conclude that each one of the files are licensed with GPL as a whole. And since all parts are interdependent, contributors own a **joint copyright over the whole project and not just their code.** Therefore, any modification to any part creates a derivative work subject to GPL and copyright laws. (An exception to this is `untar.go` which has its own license/copyright and can be conveyed separately)

Section 4 of the license states that:

> ... **keep intact all notices stating that this License and any non-permissive terms added in accord with section 7 apply to the code; keep intact all notices of the absence of any warranty; and give all recipients a copy of this License along with the Program.**

Section 5.c also adds:
>...This License gives no permission to license the work in any other way, but it does not invalidate such permission if you have separately received it.

Therefore, by modifying the license with additional terms, Lawl has failed to keep it intact, violating the license. Unless of course he has permission from other copyright holders to do so. Had he done that when he was the only copyright holder, we wouldn't be having this discussion, since he would have every right to do so.

For example, Microsoft can't take the Linux kernel, customize it to fit their needs, which includes creating an entirely new module that is based on, linking to, or using parts of the kernel, and release it under a new license or alter the existing license in any way, shape or form without infringing the copyrights of the Linux contributors and without violating the license. If that was the case, we would be seeing things like this happen every day.

If you have additional doubts, questions or need legal help, you can always bring the issue to FSF's attention at [license-violation@gnu.org](mailto:license-violation@gnu.org)

**Reply by hendursaga** [reply]
Has anyone got in touch with the FSF concerning this question? Or maybe there's a FAQ somewhere, if it does, in fact, come up this often?

**Reply by Technetium1** [reply]
@hendursaga yes - see #359

**Comment by ryleu** [comment]
Perhaps…

 - NoiseTorcher
 - EarSaver
 - BlueTorch (reference to blue switches)

**Comment by laurent-simon** [comment]
Why not one of these:

* Calmic (call/calm mic)
* Zencall (zen/then call)
* Noizero (noise zero / noise hero)
